_posts/2024-07-19-Using-Redis-with-Sidecar.md

```diff
@@ -0,0 +1,179 @@
+In Startup.cs, configure your web API to connect to the Redis instance:
+
+    ```csharp
+    public void ConfigureServices(IServiceCollection services)
+    {
+        // Other service configurations
+
+        // Add Redis
+        services.AddStackExchangeRedisCache(options =>
+        {
+            options.Configuration = Configuration.GetConnectionString("Redis");
+            options.InstanceName = "Redis.Core.WebApi_";
+        });
+
+        // Other service configurations
+    }
+    ```
+
+3. **Implementing Caching in the API**
+Whenever the API is called, it will first check if the values are in the cache. If not, it will fetch them from the database:
+
+    ```csharp
+    public async Task<IEnumerable<WeatherForecast>> GetWeatherForecasts()
+    {
+        var dataFromCache = await _cache.GetRecordAsync<IEnumerable<WeatherForecast>>(WeatherData_Key);
+        if (dataFromCache != null)
+        {
+            _logger.LogInformation("Data loaded from cache @" + DateTime.Now);
+            return dataFromCache;
+        }
+        else
+        {
+            _logger.LogInformation("Data loaded from db @" + DateTime.Now);
+            var dataFromDb = GetDataFromDB();
+            await _cache.SetRecordAsync<IEnumerable<WeatherForecast>>(dataFromDb, WeatherData_Key);
+            return dataFromDb;
+        }
+    }
+    ```
+
+4. **Adding Values to the Cache**
+We have an extension method that adds values to the cache:
+
+    ```csharp
+    public static async Task SetRecordAsync<T>(this IDistributedCache cache,
+        T value,
+        string key,
+        TimeSpan? absoluteExpiry = null,
+        TimeSpan? slidingExpiry = null)
+    {
+        // Caching option
+        var cachingOption = new DistributedCacheEntryOptions();
+
+        // Cache Time to Live, from Current Time to <specified> mins, after that cache will expire
+        cachingOption.AbsoluteExpirationRelativeToNow = absoluteExpiry ?? TimeSpan.FromSeconds(60);
+
+        // Cache Time to Live, will slide if it is accessed within the sliding window
+        cachingOption.SlidingExpiration = slidingExpiry;
+
+        // Convert to JSON
+        var dataToStore = JsonSerializer.Serialize(value);
+
+        // Set Cache
+        await cache.SetStringAsync(key, dataToStore, cachingOption);
+    }
+    ```
+
+5. **Running Your Application**
+Now you can run your application. Open your browser and navigate to:
+`http://localhost:5000/weatherforecast`.
+
+The browser will display the weather forecast in JSON format. Since we have implemented logging in the application, you will see logs in the command prompt indicating whether the response is coming from the database or the cache.
+
+## Deploying the Application to Linux App Service
+
+To get started, you'll need to containerize your .NET application. This [tutorial](https://learn.microsoft.com/en-us/dotnet/core/docker/build-container?tabs=windows&pivots=dotnet-8-0) walks you through the process step by step.
```

Comment:
remove */en-us/* from links this help route folks to localized content

Reply:
Done.